users/bcat/rules.mk

```
@@ -16,22 +19,53 @@ NKRO_ENABLE = yes
 # Enable link-time optimization to reduce binary size.
 LTO_ENABLE = yes
 
-# Disable unused build options on all keyboards.
+# Include common utilities shared across all our keymaps.
+SRC += bcat.c
+
+# Include additional utilities that extend optional QMK features only enabled
+# on some keyboards.
+ifeq ($(strip $(OLED_ENABLE)), yes)
+	SRC += bcat_oled.c
+	WPM_ENABLE = yes  # for WPM and animated "keyboard pet" widgets
+
+	# OLED pets (animated critters that react to typing) take up a lot of
+	# firmware space, so only compile one, and only if requested.
+	ifeq ($(strip $(BCAT_OLED_PET)), isda)
+		SRC += bcat_oled_pet_isda.c
+		OPT_DEFS += -DBCAT_OLED_PET
+	endif
+	ifeq ($(strip $(BCAT_OLED_PET)), luna)
+		SRC += bcat_oled_pet_luna.c
+		OPT_DEFS += -DBCAT_OLED_PET
+	endif
+endif
```

This is a bit more complicated (well, maybe not just "a bit" ...) but this should be much more flexible.  Eg, you should be able to just drop more files into the folder, without needing to change anything. 

This is definitely optional, and should be tested if applied.

```suggestion
	BCAT_OLED_PET ?= no
	ifneq ($(strip $(BCAT_OLED_PET)), no)
		ifneq (,$(wildcard $(USER_PATH)/bcat_oled_pet_$(strip $(BCAT_OLED_PET)).c))
	        SRC += $(USER_PATH)/bcat_oled_pet_$(strip $(BCAT_OLED_PET)).c
			OPT_DEFS += -DBCAT_OLED_PET
			OPT_DEFS += -DBCAT_OLED_PET_$(strip $(shell echo $(BCAT_OLED_PET) | tr '[:lower:]' '[:upper:]'))
		endif
	endif
endif
```